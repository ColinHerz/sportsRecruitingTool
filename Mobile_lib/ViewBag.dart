import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'Session.dart';
import 'dart:convert';
import 'dart:async';

class ViewBag extends StatefulWidget{
  final String bagID;

  ViewBag(this.bagID);

  _ViewBag createState() => _ViewBag(bagID);
}

class _ViewBag extends State<ViewBag>{
  final String bagID;
  // ignore: non_constant_identifier_names
  final String WOOD = "Wood";
  // ignore: non_constant_identifier_names
  final String IRON = "Iron";
  // ignore: non_constant_identifier_names
  final String WEDGE = "Wedge";
  // ignore: non_constant_identifier_names
  final String PUTTER = "Putter";
  // ignore: non_constant_identifier_names
  final String WOOD2 = "1";
  // ignore: non_constant_identifier_names
  final String IRON2 = "2";
  // ignore: non_constant_identifier_names
  final String WEDGE2 = "3";
  // ignore: non_constant_identifier_names
  final String PUTTER2 = "4";
  String bagName;
  List<Widget> display = new List<Widget>();
  List<String> clubNames = new List<String>();
  List<String> clubIDs = new List<String>();
  List<int> clubTypes = new List<int>();
  final _clubNameController = new TextEditingController();
  String _clubType;
  String _clubType2 = "Wood";

  _ViewBag(this.bagID);

  String determineClubType2() {
    switch(_clubType2)
    {
      case "Wood":
        return WOOD2;

      case "Iron":
        return IRON2;

      case "Wedge":
        return WEDGE2;

      case "Putter":
        return PUTTER2;
    }
  }
  String determineClubType(int _clubType) {

    switch(_clubType)
    {
      case 1:
        return WOOD;

      case 2:
        return IRON;

      case 3:
        return WEDGE;

      case 4:
        return PUTTER;
    }
  }

  Future<void> getBag(dynamic info) async {
    List<dynamic> clubResponse = new List<dynamic>();
    final http.Response _response = await http.post(
      'http://localhost3000.us-east-2.elasticbeanstalk.com/api/golf/getGolfBag',
      headers: headers,
      body: info
    );

    if (_response.statusCode == 200) {
      var parsedJson = json.decode(_response.body);
      clubResponse = parsedJson['golfClub'];
      setState(() {
        bagName = parsedJson['bagName'];
        for (int i = 0; i < clubResponse.length; i++) {
          clubNames.add(clubResponse[i]['clubName']);
          clubIDs.add(clubResponse[i]['_id']);
          clubTypes.add(clubResponse[i]['clubType']);
        }
      });
    }
    else {
      print(_response.statusCode);
      print(_response.body);
    }
  }
  Future<void> addClub(dynamic info) async {
    final http.Response _response = await http.post(
        'http://localhost3000.us-east-2.elasticbeanstalk.com/api/golf/createGolfclub',
        headers: headers,
        body: info
    );

    if (_response.statusCode == 200) {

    }
    else {
      print(_response.statusCode);
      print(_response.body);
    }
  }
  Future<void> deleteClub(dynamic info) async {
    final http.Response _response = await http.post(
        'http://localhost3000.us-east-2.elasticbeanstalk.com/api/golf/deleteGolfclub',
        headers: headers,
        body: info
    );

    if (_response.statusCode == 200) {
    }
    else {
      print(_response.statusCode);
      print(_response.body);
    }
  }
  Future<void> deleteBag(dynamic info) async {
    final http.Response _response = await http.post(
        'http://localhost3000.us-east-2.elasticbeanstalk.com/api/golf/deleteGolfBag',
        headers: headers,
        body: info
    );

    if (_response.statusCode == 200) {
      Navigator.pop(context);
    }
    else {
      print(_response.statusCode);
      print(_response.body);
    }
  }

  List<Widget> displayClubs(){
    display = new List<Widget>();

    display.add(Padding(
        padding: EdgeInsets.fromLTRB(0, 0, 0, 10),
        child: Text("$bagName", style: largeStyle)
    ));

    display.add(Text("Clubs in this bag:", style: TextStyle(
        fontSize: 20,
        fontWeight: FontWeight.bold
    )
    ));

    display.add(Padding(padding: EdgeInsets.fromLTRB(0, 0, 0, 20)));

    for (int i = 0; i < clubNames.length; i++)
      display.add((
          Row(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Container(
                child: InkWell(
                  onTap: () {
                  },
                  child: Text("${clubNames[i]} [${determineClubType(clubTypes[i])}]", style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold, color: Colors.white)),
                ),
                color: Colors.deepOrange,
              ),
              Padding(padding: EdgeInsets.fromLTRB(0, 0, 20, 0)),
              ButtonTheme(
                minWidth: 60,
                height: 30,
                child: RaisedButton(
                  onPressed: (){
                      deleteClub(jsonEncode(<String, String>{
                        'golfClub': clubIDs[i],
                        'golfBag': bagID,
                      }));
                  },
                  child: Text("Delete"),
                )
              )
            ],
          )
      ));

    display.add(RaisedButton(
        onPressed: (){
          clubby(context);
        },
        child: Text('Add Club')
    ));

    display.add(RaisedButton(
        onPressed: (){
          deleteBag(jsonEncode(<String, String>{
            'golfBag': bagID,
          }));
        },
        child: Text('Delete Bag')
    ));

    return display;
  }

  AlertDialog clubby(BuildContext context){
    showDialog(
        context: context,
        builder: (BuildContext context){
          return AlertDialog(
            title: Text('Club', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
            content: SingleChildScrollView(
                child: ListBody(
                  children: <Widget>[
                    TextField(
                        controller: _clubNameController,
                        decoration: InputDecoration(hintText: "Club Name")
                    ),
                    Padding(
                      padding: EdgeInsets.fromLTRB(0, 20, 0, 0),
                      child: Text("Club type"),
                    ),
                    DropdownButton<String>(
                      value: _clubType2,
                      icon: Icon(Icons.arrow_downward),
                      iconSize: 24,
                      elevation: 16,
                      style: TextStyle(color:Colors.deepOrange),
                      onChanged: (String newValue) {
                        setState(() {
                          _clubType2 = newValue;
                        });
                      },
                      items: <String> ['Wood', 'Iron', 'Wedge', 'Putter'].map<DropdownMenuItem<String>>((String value) {
                        return DropdownMenuItem<String>(
                          value: value,
                          child: Text(value, style: TextStyle(fontSize: 18)),
                        );
                      }).toList(),
                    ),
                  ],
                )
            ),
            actions: <Widget>[
              new FlatButton(
                  child: new Text('Add Club'),
                  onPressed: () {
                    addClub(jsonEncode(<String, String>{
                      'golfBag' : bagID,
                      'clubName' : _clubNameController.text,
                      'clubType' : determineClubType2(),
                    }));
                    Navigator.of(context).pop();
                  }
              ),
              new FlatButton(
                  child: new Text('Cancel'),
                  onPressed: () {
                    Navigator.of(context).pop();
                  }
              )
            ],
          );
        }
    );
  }

  void initState(){
    super.initState();
    getBag(jsonEncode(<String, String>{
      'golfBag': bagID,
    }));
  }

  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('View Bag'),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.start,
          children: displayClubs(),
        ),
      ),
    );
  }
}
