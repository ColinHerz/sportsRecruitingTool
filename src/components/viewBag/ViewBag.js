import apiCall from "../../api/api.js";
import PropTypes from "prop-types";
import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";

import "./viewBag.scss";

const NO_CLUB = 0;
const WOOD = 1;
const IRON = 2;
const WEDGE = 3;
const PUTTER = 4;

const ViewBag = props => {
	const { bid } = useParams();

	const [showError, setShowError] = useState(false);

	const [bag, setBag] = useState({ golfClub: [], bagName: `Loading...` });
	const [updateBag, setUpdateBag] = useState(true);

	const [addClub, setAddClub] = useState(false);
	const [clubName, setClubName] = useState(``);
	const [clubType, setClubType] = useState(NO_CLUB);

	useEffect(() => {
		if (updateBag) {
			apiCall(
				{
					endpoint: `/golf/getGolfBag`,
					type: `POST`,
					body: {
						golfBag: bid
					}
				},
				data => {
					if (data.status !== 200) {
						setShowError(true);
						return;
					}

					setBag(JSON.parse(data.response));
				},
				apiError
			);

			setUpdateBag(false);
		}
	});

	const submitClub = event => {
		event.preventDefault();

		if (clubType === NO_CLUB) {
			setShowError(true);
			return;
		}

		apiCall(
			{
				endpoint: `/golf/createGolfclub`,
				type: `POST`,
				body: {
					golfBag: bid,
					clubName: clubName,
					clubType: clubType
				}
			},
			apiSuccess,
			apiError
		);

		setAddClub(false);
		setClubName(``);
		setClubType(NO_CLUB);
	};

	const handleDeleteClub = event => {
		event.preventDefault();

		const clubId = event.target.parentElement.id;

		const body = {
			golfClub: clubId,
			golfBag: bid
		};

		apiCall(
			{
				endpoint: `/golf/deleteGolfclub`,
				type: `POST`,
				body: {
					golfClub: clubId,
					golfBag: bid
				}
			},
			apiSuccess,
			apiError
		);
	};

	const apiSuccess = data => {
		if (data.status !== 200) {
			setShowError(true);
			return;
		}

		setUpdateBag(true);
	}

	const apiError = () => {
		setShowError(true);
	}

	const cancelAddClub = event => {
		event.preventDefault();

		setAddClub(false);
		setClubName(``);
		setClubType(NO_CLUB);
	};

	const handleAddClubClick = event => {
		event.preventDefault();

		setAddClub(true);
	};

	const handleChangeClubName = event => {
		event.preventDefault();

		setClubName(event.target.value);
	};

	const handleChangeClubType = event => {
		event.preventDefault();

		setClubType(parseInt(event.target.value));
	};

	return (
		<main id="view-bag">
			{
				showError ?
					<p id="error-message">An error occurred. Please try again later.</p>:
					null
			}

			<h2>{bag.bagName}</h2>

			<ul>
				{
					bag.golfClub.map(club => {
						let typeName = ``;

						switch(club.clubType) {
						case WOOD:
							typeName = `Wood`;
							break;
						case IRON:
							typeName = `Iron`;
							break;
						case WEDGE:
							typeName = `Wedge`;
							break;
						case PUTTER:
							typeName = `Putter`;
							break;
						default:
							console.error(`Incorrect putter type`);
							setShowError(true);
							break;
						}

						return (
							<li key={club._id} id={club._id}>
								<p>{club.clubName} <span className="gray-type">{typeName}</span></p>
								<button onClick={handleDeleteClub}>Delete</button>
							</li>
						);
					})
				}
			</ul>

			{
				addClub ?
					<div id="add-club-div">
						<label>
							Name:
							<input
								type="text"
								onChange={handleChangeClubName}
								value={clubName}
							/>
						</label>

						<label>
							Type:
							<select
								value={clubType}
								onChange={handleChangeClubType}
							>
								<option value={NO_CLUB}>Please choose a club type</option>
								<option value={WOOD}>Wood</option>
								<option value={IRON}>Iron</option>
								<option value={WEDGE}>Wedge</option>
								<option value={PUTTER}>Putter</option>
							</select>
						</label>

						<div>
							<button onClick={submitClub}>Add</button>
							<button onClick={cancelAddClub}>Cancel</button>
						</div>
					</div>:
					null
			}

			<button
				onClick={handleAddClubClick}
				id="add-club-btn"
			>
				Add Club
			</button>
		</main>
	);
};

ViewBag.propTypes = {
	user: PropTypes.object.isRequired
};

export default ViewBag;
