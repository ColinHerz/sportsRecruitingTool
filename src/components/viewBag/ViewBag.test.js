/* eslint-env jest */
/*eslint no-unused-vars: ["error", { "varsIgnorePattern": "apiCall" }]*/

import apiCall from "../../api/api.js";
import { MemoryRouter } from "react-router-dom";
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "../../../enzyme.js";

import ViewBag from "./ViewBag.js";

const genProps = () => {
	return {
		user: {
			id: `garbage`,
			email: `john.smith@yahoo.net`,
			firstname: `John`,
			lastname: `Smith`,
			isVerified: true
		}
	};
};

beforeEach(() => {
	jest.resetModules();
});

describe(`Rendering`, () => {
	it(`Renders correctly`, () => {
		const props = genProps(false);

		const component = renderer.create(
			<MemoryRouter>
				<ViewBag {...props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
