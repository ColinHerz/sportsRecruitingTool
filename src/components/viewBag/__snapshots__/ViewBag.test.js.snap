// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Renders correctly 1`] = `
<main
  id="view-bag"
>
  <h2>
    Loading...
  </h2>
  <ul />
  <button
    id="add-club-btn"
    onClick={[Function]}
  >
    Add Club
  </button>
</main>
`;
